metrics-core/src/main/java/com/codahale/metrics/MetricRegistry.java

```diff
@@ -83,10 +83,15 @@ public MetricRegistry() {
      * @param metric the metric
      * @param <T>    the type of the metric
      * @return {@code metric}
-     * @throws IllegalArgumentException if the name is already registered
+     * @throws IllegalArgumentException if the name is already registered or metric variable is null
      */
     @SuppressWarnings("unchecked")
     public <T extends Metric> T register(String name, T metric) throws IllegalArgumentException {
+
+        if (metric == null) {
+            throw new IllegalArgumentException("metric variable must not be null !!!");
```

Comment:
I'd make this a `throw new NullPointerException("metric == null");` instead

Reply:
Fixed as request